internal/backend/runtime/omni/migration/migrations.go

```diff
@@ -1537,3 +1538,116 @@ func moveEtcdBackupStatuses(ctx context.Context, st state.State, logger *zap.Log
 
 	return nil
 }
+
+// createVersionContractRevertConfigPatch creates a config patch to prevent the unwanted reboot
+// which would be caused by the following version contract fix: https://github.com/siderolabs/omni/issues/1097
+//
+// We create a system config patch with a low weight (so that the user patches will override it)
+// to preserve the config changes the revert would cause which would require a reboot, namely:
+// - preserve machine.features.diskQuotaSupport setting
+// - preserve machine.features.apidCheckExtKeyUsage setting.
+func createVersionContractRevertConfigPatch(ctx context.Context, st state.State, logger *zap.Logger) error {
+	clusterConfigVersionList, err := safe.StateListAll[*omni.ClusterConfigVersion](ctx, st)
+	if err != nil {
+		return fmt.Errorf("failed to list cluster config versions: %w", err)
+	}
+
+	clusterConfigVersionMap := make(map[string]string, clusterConfigVersionList.Len())
+	for clusterConfigVersion := range clusterConfigVersionList.All() {
+		clusterConfigVersionMap[clusterConfigVersion.Metadata().ID()] = clusterConfigVersion.TypedSpec().Value.Version
+	}
+
+	clusterMachineConfigList, err := safe.StateListAll[*omni.ClusterMachineConfig](ctx, st)
```

Comment:
one thing to note here is that this is the last generated config, not the one last applied yet, but I don't want to overcomplicate this migration even further

Reply:
discussed with @Unix4ever and changed it to check the last applied config version, skip if it is not up-to-date. IT is not 100% buletproof, but probably will give better results (particularly for disconnected machines).